src/ledc.rs

```diff
@@ -236,19 +221,50 @@ impl<'d> LedcDriver<'d> {
             }
         }
 
-        // SAFETY: As long as we have borrowed the timer, we are safe to use
-        // it.
-        esp!(unsafe { ledc_channel_config(&channel_config) })?;
-
-        Ok(LedcDriver {
-            duty,
-            hpoint,
+        let mut driver = LedcDriver {
+            duty: 0,
+            hpoint: 0,
             speed_mode: timer_driver.borrow().speed_mode,
             max_duty: timer_driver.borrow().max_duty,
             timer: timer_driver.borrow().timer() as _,
             channel: C::channel() as _,
+            pin: 0,
             _p: PhantomData,
-        })
+        };
+
+        driver.set_pin(pin)?;
+
+        driver.apply_config()?;
+
+        Ok(driver)
+    }
+
+    pub fn apply_config(&mut self) -> Result<(), EspError> {
```

Comment:
any specific reason why you want the apply_config to be a pub function? Since its invoked on creation with new and on set_pin i dont see why it does need to be a separate public method.

Reply:
indeed it should not be pub